browser_patches/webkit/patches/bootstrap.diff

```diff
@@ -7445,19 +8238,63 @@ index e798e912e47b0d8d0f038e7f76d70fc06af6ecae..20ac77835f2453ad0cf5f229aeeb158f
  }
  
  bool ArgumentCoder<WindowFeatures>::decode(Decoder& decoder, WindowFeatures& windowFeatures)
-@@ -1412,6 +1415,12 @@ bool ArgumentCoder<WindowFeatures>::decode(Decoder& decoder, WindowFeatures& win
+@@ -1412,6 +1419,12 @@ bool ArgumentCoder<WindowFeatures>::decode(Decoder& decoder, WindowFeatures& win
+         return false;
+     if (!decoder.decode(windowFeatures.dialog))
+         return false;
++    if (!decoder.decode(windowFeatures.noopener))
++        return false;
++    if (!decoder.decode(windowFeatures.noreferrer))
++        return false;
++    if (!decoder.decode(windowFeatures.additionalFeatures))
++        return false;
+     return true;
+ }
+ 
+@@ -1425,6 +1438,9 @@ void ArgumentCoder<DragData>::encode(Encoder& encoder, const DragData& dragData)
+ #if PLATFORM(COCOA)
+     encoder << dragData.pasteboardName();
+     encoder << dragData.fileNames();
++#endif
++#if PLATFORM(WPE)
++    encoder << *dragData.platformData();
+ #endif
+     encoder << dragData.dragDestinationActionMask();
+ }
+@@ -1447,10 +1463,18 @@ bool ArgumentCoder<DragData>::decode(Decoder& decoder, DragData& dragData)
+     if (!decoder.decode(applicationFlags))
+         return false;
+ 
+-    String pasteboardName;
+-    Vector<String> fileNames;
++#if PLATFORM(WPE)
++    SelectionData* selectionData = new SelectionData();
++    if (!decoder.decode(*selectionData))
++        return false;
++    SelectionData* firstArg = selectionData;
++#else
++    String firstArg;
++#endif
++
+ #if PLATFORM(COCOA)
+-    if (!decoder.decode(pasteboardName))
++    Vector<String> fileNames;
++    if (!decoder.decode(firstArg))
          return false;
-     if (!decoder.decode(windowFeatures.dialog))
+ 
+     if (!decoder.decode(fileNames))
+@@ -1461,8 +1485,10 @@ bool ArgumentCoder<DragData>::decode(Decoder& decoder, DragData& dragData)
+     if (!decoder.decode(dragDestinationActionMask))
          return false;
-+    if (!decoder.decode(windowFeatures.noopener))
-+        return false;
-+    if (!decoder.decode(windowFeatures.noreferrer))
-+        return false;
-+    if (!decoder.decode(windowFeatures.additionalFeatures))
-+        return false;
+ 
+-    dragData = DragData(pasteboardName, clientPosition, globalPosition, draggingSourceOperationMask, applicationFlags, dragDestinationActionMask);
++    dragData = DragData(firstArg, clientPosition, globalPosition, draggingSourceOperationMask, applicationFlags, dragDestinationActionMask);
++#if PLATFORM(COCOA)
```

Comment:
Isn't it used on Windows too?

Reply:
Windows doesn't have drag & drop support in webkit2, so it never gets used. My follow up patch adds windows support.